scenarios/kubernetes_e2e.py

```diff
@@ -0,0 +1,304 @@
+
+
+def continuation_lines(fin):
+    """Reads in lines respect \\ in the end."""
+    for line in fin:
+        line = line.rstrip('\n')
+        while line.endswith('\\'):
+            line = line[:-1] + next(fin).rstrip('\n')
+        yield line
+
+
+def load_env(envpath):
+    """Load a env file into os.environ."""
+    if not os.path.isfile(envpath):
+        print >>sys.stderr, 'envfile %s does not exist' % envpath
+        return
+    with open(envpath, 'r') as envfile:
+        for line in continuation_lines(envfile):
+            tup = line.strip().split('=', 1)
+            if len(tup) == 2:
+                os.environ[tup[0].strip()] = tup[1].strip()
+
+
+def load_template(template_path, args):
+    """Load a templated env file into os.environ"""
+
+    env_map = {
+        '${version_client}' : args.version_client,
+        '${version_cluster}' : args.version_cluster,
+        '${version_old}' : args.version_old,
+        '${version_new}' : args.version_new,
+        '${image_old}' : args.image_old,
+        '${image_new}' : args.image_new
+    }
+
+    if not os.path.isfile(template_path):
+        print >>sys.stderr, 'envfile %s does not exist' % template_path
+        return
+    with open(template_path, 'r') as envfile:
+        for line in continuation_lines(envfile):
+            # Resolve vars
+            for key, val in env_map.items():
+                if key in line and val:
+                    line = line.replace(key, val)
+            tup = line.strip().split('=', 1)
+            if len(tup) == 2:
+                os.environ[tup[0].strip()] = tup[1].strip()
+
+
+def sigterm_handler(_signo, _frame):
+    """Handle signal.SIGTERM."""
+    print 'SIGTERM signo = %s, frame = %s' % (_signo, _frame)
+    check(['docker', 'stop', CONTAINER])
+
+
+def sigint_handler(_signo, _frame):
+    """Handle signal.SIGINT."""
+    print 'SIGINT signo = %s, frame = %s' % (_signo, _frame)
+    check(['docker', 'stop', CONTAINER])
+
+
+def main(args):
+    """Set up env, trigger runner, handle termination. """
+    # pylint: disable=too-many-locals
+    # pylint: disable=too-many-branches
+    # pylint: disable=too-many-statements
+
+    # platform envs
+    if args.platform:
+        load_env(test_infra('platforms/' + args.platform + '.env'))
+
+    # job envs
+    if args.env_file:
+        load_env(test_infra('jobs/' + args.env_file + '.env'))
+
+    # job env templates
+    if args.env_template:
+        load_template(test_infra('jobs/' + args.env_template + '.env'), args)
+
+    # Other args Override
+    if args.zone:
+        os.environ['ZONE'] = args.zone
+        os.environ['KUBE_GCE_ZONE'] = args.zone
+    if args.project:
+        os.environ['PROJECT'] = args.project
+    if args.use_cert_auth:
+        os.environ['CLOUDSDK_CONTAINER_USE_CLIENT_CERTIFICATE'] = 'true'
+
+    # Boilerplate envs
+    # Assume we're upping, testing, and downing a cluster
+    os.environ['E2E_UP'] = os.environ.get('E2E_UP') or 'true'
```

Comment:
Consider dict.setdefault() for this

Also, I would be much happier if we simplify this to set the variables explicitly rather than relying on magic branching behavior.